src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -2614,6 +2562,39 @@ export class StudyViewPageStore {
             .value();
     }
 
+    readonly genePanelData = remoteData({
+        await: () => [this.molecularProfiles, this.samples],
+        invoke: async () => {
+            const studyMolecularProfilesSet = _.groupBy(
+                this.molecularProfiles.result,
+                molecularProfile => molecularProfile.studyId
+            );
+
+            const sampleMolecularIdentifiers = _.flatMap(
+                this.samples.result,
+                sample => {
+                    return studyMolecularProfilesSet[sample.studyId].map(
+                        molecularProfile => {
+                            return {
+                                molecularProfileId:
+                                    molecularProfile.molecularProfileId,
+                                sampleId: sample.sampleId,
+                            };
+                        }
+                    );
+                }
+            );
+
+            const genePanelData = await defaultClient.fetchGenePanelDataInMultipleMolecularProfilesUsingPOST(
+                {
+                    sampleMolecularIdentifiers: sampleMolecularIdentifiers,
+                }
+            );
+            return genePanelData.filter(datum => datum.profiled);
```

Comment:
should the fact that we are only returning profiled data be reflected in the name of this promise?  should be have two promises, one with raw data, one with filtered?

Reply:
We just need profiled data. I'll rename the promise